File: day13/multiplication_table.py
def print_multiplication_table():
    while True:
        user_input = input("Please enter a number and I will print\nthe multiplication table, to 10, for that number: ")
        try:
            number = int(user_input)
            for i in range(1, 11):
                print(f"{number} x {i} = {number * i}")
            break    
        except ValueError:
            print("\nPlease enter a number.\n")

if __name__ == "__main__":
    print_multiplication_table()

File: day13/MultiplicationTable.java
import java.util.Scanner;

public class MultiplicationTable {
    public static void main(String[] args) {
        printMultiplicationTable();
    }

    public static void printMultiplicationTable() {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.print("Please enter a number and I will print\nthe multiplication table, to 10, for that number: ");
            String input = scanner.nextLine();
            System.out.println(); // extra line after prompt

            try {
                int number = Integer.parseInt(input);
                for (int i = 1; i <= 10; i++) {
                    System.out.println(number + " x " + i + " = " + (number * i));
                }
                break;
            } catch (NumberFormatException e) {
                System.out.println("\nPlease enter a number.\n");
            }
        }

        scanner.close();
    }
}
